analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -57,23 +58,29 @@ public abstract class UseUnixEpochBase<TSyntaxKind, TLiteralExpression, TMemberA
                 Language.GeneratedCodeRecognizer,
                 c =>
                 {
-                    var literalsArguments = Language.Syntax.ArgumentExpressions(c.Node).OfType<TLiteralExpression>();
-                    if (!literalsArguments.Any(x => IsValueEqualTo(x, EpochYear) || literalsArguments.Count(x => IsValueEqualTo(x, EpochMonth)) != 2))
+                    var arguments = Language.Syntax.ArgumentExpressions(c.Node);
+                    var literalsArguments = arguments?.OfType<TLiteralExpression>();
+
+                    if (literalsArguments.Any(x => IsValueEqualTo(x, EpochYear)
+                        && literalsArguments.Count(x => IsValueEqualTo(x, EpochMonth)) == 2)
+                        && CheckAndGetTypeName(c.Node, c.SemanticModel, out var name)
+                        && IsEpochCtor(c.Node, c.SemanticModel))
                     {
-                        return;
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), name));
                     }
-
-                    var type = c.SemanticModel.GetTypeInfo(c.Node).Type;
-                    if (type.IsAny(TypesWithUnixEpochField) && IsEpochCtor(c.Node, c.SemanticModel))
+                    else if (arguments.Count() == 1
+                        && ((literalsArguments.Count() == 1  && IsValueEqualTo(literalsArguments.First(), EpochTicks))
```

Comment:
This can also be done inside the `IsEpochCtor` method:

`|| IsParameterExistingAndLiteralEqualTo("ticks", EpochYear, lookup)`